### PR TITLE
[telemetrygen] use log sdk instead of pdata

### LINKED ISSUE #18902: [cmd/telemetrygen] migrate log to use go sdk
### Component(s)

cmd/telemetrygen

### Is your feature request related to a problem? Please describe.

For now, since the log of go sdk is now frozen. So the telemetrygen can only use `pdata` and other package to create log data.

When go log sdk is ready, we could check whether it is sufficient for telemetrygen to generate log data.

### Describe the solution you'd like

None

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by codeboten**
This updates telemetrygen to use the otel go SDK for logs instead of pdata for the logging signal. This aligns with the tracing and metrics signal.

Fixes #18902

### CODE REVIEW

File: cmd/telemetrygen/internal/logs/worker.go [hunk 1]
```diff
@@ -10,78 +10,95 @@ import (
...
+			Attributes:        attrs,
+			TraceID:           tid,
+			SpanID:            sid,
+			Resource:          res,
+			DroppedAttributes: 1,
+			// ObservedTimestamp time.Time
+			// TraceFlags        trace.TraceFlags
+			// InstrumentationScope *instrumentation.Scope
+			// AttributeValueLengthLimit int
+			// AttributeCountLimit       int
```
Comment (mwear) [inline cmd/telemetrygen/internal/logs/worker.go:92]: Is there a reason to leave these lines commented out or can we delete?

Comment (mwear) [inline cmd/telemetrygen/internal/logs/worker.go:92]:
```suggestion
```

Comment (codeboten) [reply cmd/telemetrygen/internal/logs/worker.go:92]: I was leaving these here are fields we may want to add details about in the future, but they can likely be removed at this point

**Review by mwear — APPROVED** [review]

File: cmd/telemetrygen/internal/logs/worker.go [hunk 2]
```diff
@@ -10,78 +10,90 @@ import (
...
-	limitPerSecond rate.Limit          // how many logs per second to generate
-	wg             *sync.WaitGroup     // notify when done
-	logger         *zap.Logger         // logger
-	index          int                 // worker index
-	traceID        string              // traceID string
-	spanID         string              // spanID string
+	running        *atomic.Bool    // pointer to shared flag that indicates it's time to stop the test
+	numLogs        int             // how many logs the worker has to generate (only when duration==0)
+	body           string          // the body of the log
+	severityNumber log.Severity    // the severityNumber of the log
```
Comment (atoulme) [inline cmd/telemetrygen/internal/logs/worker.go:27]: Is this the same number range as before?

Comment (codeboten) [reply cmd/telemetrygen/internal/logs/worker.go:27]: yup

**Pull request merged**